packages/core/src/modules/connections/ConnectionsApi.ts

```diff
@@ -278,6 +287,57 @@ export class ConnectionsApi {
     return message
   }
 
+  public async rotate(options: { connectionId: string; did?: string; routing?: Routing }) {
+    const { connectionId, did, routing } = options
+    const connection = await this.connectionService.getById(this.agentContext, connectionId)
+
+    if (did && routing) {
+      throw new AriesFrameworkError(`'routing' is disallowed when defining 'did'`)
+    }
+
+    const message = await this.didRotateService.createRotate(this.agentContext, {
+      connection,
+      did,
+      routing: routing || (await this.routingService.getRouting(this.agentContext, {})),
```

Comment:
Do we also want to create routing when we have a toDid specified?

Reply:
If `toDid` is specified, routing is disallowed (the same thing as we did in #1550).

Reply:
Yes if routing is not provided we will go to `|| (await this.routingService.getRouting(this.agentContext, {})),`, so it willl still be generated right? I think we shouldn't call `getRouting` in this case

Reply:
I'm under a heavy jetlag right now but I think that case will not happen, as a few lines before that we have:

```ts
if (toDid && routing) {
      throw new AriesFrameworkError(`'routing' is disallowed when defining 'did'`)
    }
```

Reply:
Yes but if `routing` is not defined, this will not throw an error. Then later on it will do `routing || (await this.routingService.getRouting(this.agentContext, {}))`. Routing will never be defined when `toDid` is defined, so that will mean it will call `this.routingService`.

I think `|| (await this.routingService.getRouting(this.agentContext, {}))` should ONLY be called if `toDid` is not defined.

```
if (toDid && routing) {
  throw new AriesFrameworkError(`'routing' is disallowed when defining 'did'`)
}

let routingToUse = routing
if (!toDid && !routing) {
  routingToUse = await this.routingService.getRouting(this.agentContext, {})
}
```

Reply:
Maybe I'm missing something here, but I think your influenced by the jetlag 😅

Reply:
Ooh I see. It is creating an unnecessary routing 🤦. Thanks for the explanation, I think I should have some rest 😄.